### PR TITLE
build frontend based on bzl artifact 

### PR DESCRIPTION
**Pull request by jhchabran**
Introduce a conditional path to build both frontends using Bazel produced artifacts instead of the normal ones. 

This is achieved by introducing a build setting named `//:assets_bundle_type` which can take either `oss` or `enterprise`. When enabled, this settings flips which files are included in the filegroup that is embedded in the `//ui/assets:assets` go library. It also automatically add the `dist` go tags to the frontend binaries, which in turns pick the right sources in the `ui/assets` Go package. 

If it's not set, it defaults to empty, which will instead run the dev variant of the assets, expecting the files to be maintained on the FS during the runtime, usually by `bundle-dev`. 

This has a side effect of invalidating the analysis cache, but it's fairly acceptable at this point. In the future, we want to refactor the assets handling code to instead have two explicit dependencies whenever that oss and enterprise pick accordingly, making the dependency graph static regardless of the bundle type. 

## Test plan

This was tested by running `sg start oss-bazel` and manually specifying the right `--//:assets_bundle_type=X` to ensure that we get the correct assets and that it stil works without it, in a classic dev setting.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.03 kb) | 0.01% (+1.77 kb) | 0.01% (+1.74 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/dtjh/bzl/docker-frontend/compare-report.html;tab=live_object) for a full comparison between the commits 547b5db and 686b21d or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).

**Note:** We do not have exact data for ecd14e6. So we have used data from: 686b21d.
The intended commit has no frontend pipeline, so we chose the last commit with one before it.

<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: .bazelrc [hunk 1]
```diff
@@ -17,3 +17,7 @@ try-import %workspace%/.aspect/bazelrc/user.bazelrc
 
 # Enable bazel hack for autogold; apply to both build & test to avoid busting analysis cache
 build --test_env=ENABLE_BAZEL_PACKAGES_LOAD_HACK=true
+
+# Force bundle settings depending on targets.
```
Comment (jhchabran) [inline .bazelrc:21]: Just left there as some sort of documentation.

File: BUILD.bazel [hunk 2]
```diff
@@ -216,3 +216,29 @@ nogo(
...
 exports_files(["go.mod"])
+
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+
+# This settings enable to switch from one compiled "dist" bundle type to the other.
+# Enabling one or the other will enable the `dist` go tag, which will reflect on the build
+# constraints of the //ui/assets package.
+#
+# If none is set, this will default back to the dev bundle.
+string_flag(
```
Comment (jhchabran) [inline BUILD.bazel:227]: This is the setting definition.

File: ui/assets/load_manifest_dist.go [hunk 3]
```diff
@@ -22,7 +23,14 @@ var (
 // details.
 func LoadWebpackManifest() (*WebpackManifest, error) {
 	webpackManifestOnce.Do(func() {
-		manifestContent, err := assetsFS.ReadFile("webpack.manifest.json")
+		f, err := afs.Open("webpack.manifest.json")
```
Comment (jhchabran) [inline ui/assets/load_manifest_dist.go:26]: We have to do this because it's now a `fs.FS` and not an `embed.FS`.

**Review by burmudar — APPROVED** [review]

**Pull request merged**